### PR TITLE
feat(integrations): Toggle features at organization integration level

### LINKED ISSUE #70316: Slack alerts in threads: add option to enable or disable it
### Problem Statement

With the addition of Slack threads for metric alerts, when the same issue triggers an alert multiple times all messages are in the same thread. 

[Some users like this behaviour](https://sentry.zendesk.com/agent/tickets/120760), some want to [see all messages in the channel ](https://github.com/getsentry/sentry/issues/70135). 

[A recent change made ](https://github.com/getsentry/sentry/pull/70108)all messages to also be posted in the channel, not only in the thread. 

### Solution Brainstorm

Make this behaviour customisable:
By default send only to thread with the option to also send to channel, or vice-versa. 

### Product Area

Settings - Integrations

### PR DESCRIPTION
**Pull request by ykamo001**
We need to allow users to be able to toggle features for integrations at their organization integration level. If users do not like or want the features, they should have the power to turn them on or off at their disposal. 

The previous FE implementation was doing this at the organization model level, but we can leverage the organization integration model/object to hold and control these settings. Added a custom component that will handle doing this toggling across the integration, and made it integration agnostic to be used across all the other integrations as well.

We can slowly migrate the github piece to this functionality. There is more work to be done on the FE side to help improve this code area, but it would require a lot of effort and changes that we can follow up with after better planning.

https://github.com/getsentry/sentry/assets/5581484/1213ca25-19d7-417e-89f2-166aada6b7ec

Requires: https://github.com/getsentry/sentry/pull/70620

### CODE REVIEW

File: src/sentry/integrations/slack/integration.py
Comment (ykamo001) [inline src/sentry/integrations/slack/integration.py:1]: merge conflict problem, removing this

**Comment by cathteng** [comment]
should we be using Form and JsonForm? it would keep the styling that's already present in the Github features page
https://github.com/getsentry/sentry/blob/1a913b72556f24a9def277b85d5c672eec772c96/static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx#L388-L405

**Comment by ykamo001** [comment]
> should we be using Form and JsonForm? it would keep the styling that's already present in the Github features page
> 
> https://github.com/getsentry/sentry/blob/1a913b72556f24a9def277b85d5c672eec772c96/static/app/views/settings/organizationIntegrations/integrationDetailedView.tsx#L388-L405

We can't use that because of the way we're calling the API to update all configs as a global state

File: static/app/views/settings/organizationIntegrations/featureToggle.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        toggleableFlags: featureToggles,
+      };
+
+      try {
+        api.request(integrationConfigEndpoint, {
+          method: 'POST',
+          data: updatedConfigData,
+        });
+      } catch (error) {
+        // Nothing to do here
```
Comment (sentaur-athena) [inline static/app/views/settings/organizationIntegrations/featureToggle.tsx:57]: What kind of error can happen here? If the toggle is not successful we should reflect that somehow in the UI. Throw a notice maybe?

Comment (ykamo001) [reply static/app/views/settings/organizationIntegrations/featureToggle.tsx:57]: If the call fails, we'll have to hold onto all the errors and show that correctly. I think at this point, there is no error handling on this component, so we may have to introduce that. Maybe a banner per error, or just a singular error?

**Comment by ykamo001** [comment]
Closing this in favor of following existing convention to keep UI the same

**Pull request closed without merging**